narwhals/translate.py

```diff
@@ -864,18 +873,11 @@ def to_py_scalar(scalar_like: Any) -> Any:
     if pd and isinstance(scalar_like, pd.Timedelta):
         return scalar_like.to_pytimedelta()
 
-    all_scalar_types = (
-        int,
-        float,
-        complex,
-        bool,
-        bytes,
-        str,
-        datetime,
-        timedelta,
-        numbers.Number,
-    )
-    if isinstance(scalar_like, all_scalar_types):
+    # SIM101 suggests merging the tuples, but that would create a new tuple,
+    # so doesn't seem ideal perf-wise?
+    if isinstance(scalar_like, NON_TEMPORAL_SCALAR_TYPES) or isinstance(  # noqa: SIM101
+        scalar_like, (datetime, timedelta)
```

Comment:
I am ok with removing the check for numbers.Number, yet, was it intended?

Also, what's the reason to keep the check for non temporal types if they return early at the beginning?

Reply:
😄 sometimes you just gotta review on the go

---

yeah if we have int, float, complex, then I'm not aware of what number.Number adds

furthermore, we have:
```python
In [10]: isinstance(np.int64(1), numbers.Number)
Out[10]: True
```
so checking `number.Number` early on would cause numpy scalars to not be converted to Python scalars

---

`pd.Timestamp` and `pd.Timedelta` would satisfy `isinstance(..., (datetime, timedelta))`, despite not being Python scalars, so if we want to do a cheap early return I think we need to exclude them

Reply:
Yep, super on go.

Got it for numbers! 👌

For the other comment I meant, why is it needed to have `isinstance(scalar_like, NON_TEMPORAL_DTYPES)` before checking for temporal in line above?

Reply:
ah right, all the branches above just return, I thought something was modifying the input and passing it down, thanks!